tools/interu/src/config/mod.rs

```diff
@@ -0,0 +1,186 @@
+use std::{collections::BTreeMap, fmt::Display, path::Path};
+
+use rand::{distributions::WeightedIndex, prelude::Distribution as _, thread_rng};
+use serde::{Deserialize, Serialize};
+use snafu::{OptionExt, ResultExt, Snafu};
+use tracing::instrument;
+
+use crate::{
+    config::{
+        profile::{Profile, StrategyValidationError, TestRun},
+        runner::{ConvertNodeGroupError, Distribution, ReplicatedNodeGroup, Runner},
+    },
+    instances::Instances,
+};
+
+pub mod profile;
+pub mod runner;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("failed to read file"))]
+    ReadFile { source: std::io::Error },
+
+    #[snafu(display("failed to deserialize file"))]
+    Deserialize { source: serde_yaml::Error },
+
+    #[snafu(display("failed to validate file"))]
+    Validate { source: ValidationError },
+
+    #[snafu(display("failed to find profile named {profile_name:?}"))]
+    UnknownProfileName { profile_name: String },
+
+    #[snafu(display("failed to convert node-group to Replicated format"))]
+    ConvertNodeGroup { source: ConvertNodeGroupError },
+}
+
+#[derive(Debug, Snafu)]
+pub enum ValidationError {
+    #[snafu(display("invalid profile config"))]
+    InvalidProfileConfig { source: StrategyValidationError },
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Config {
+    #[serde(default, skip_serializing_if = "BTreeMap::is_empty")]
+    pub runners: BTreeMap<String, Runner>,
+
+    #[serde(default, skip_serializing_if = "BTreeMap::is_empty")]
+    pub profiles: BTreeMap<String, Profile>,
+}
+
+impl Config {
+    #[instrument(name = "load_config_from_file", skip(path), fields(path = %path.as_ref().display()))]
+    pub fn from_file<P>(path: P) -> Result<Self, Error>
+    where
+        P: AsRef<Path>,
+    {
+        let contents = std::fs::read_to_string(path).context(ReadFileSnafu)?;
+
+        tracing::debug!("deserialize file contents");
+        let config: Config = serde_yaml::from_str(&contents).context(DeserializeSnafu)?;
```

Comment:
Just an idea, use `Self` to make it clear that Config isn't something else.

```suggestion
        let config: Self = serde_yaml::from_str(&contents).context(DeserializeSnafu)?;
```